### PR TITLE
allow manual registration of mvc routes for openapi

### PR DESCRIPTION
**Pull request by tpoll**
@jknack  as discussed in https://github.com/jooby-project/jooby/issues/3059, this adds support to manually register a mvc class for open-api generation.

In my testing, order of the  `registerMvc` calls didn't break anything, just change the order of the generated paths in the `openapi.json` file.

**Open problem**:
Right now this doesn't actually scope the registration to a particular router.  Open to any ideas on how we could make that work. I thought about adding another param to `registerMvc`, but I didn't see an obvious way in code to match the value of that param to the owning routers in the parsing logic.

Very open to any feedback!

### CODE REVIEW

**Comment by jknack** [comment]
we need a test that demonstrate how this works and helps (success case).

> Open problem: Right now this doesn't actually scope the registration to a particular router. Open to any ideas on how we could make that work. I thought about adding another param to registerMvc, but I didn't see an obvious way in code to match the value of that param to the owning routers in the parsing logic.

Another test that shows what the issue is.

**Comment by tpoll** [comment]
Okay, I believe I have added tests that properly demonstrate the issue and how this works to resolve it.

I believe the `open problem` I earlier mentioned isn't actually an issue, since the openapi generator seems to always be scoped to a single jooby application class. I also added a test to demonstrate this.

**Comment by jknack** [comment]
Thank you. It is clear now how it works. What if we add a custom annotation with clear documentation of when/how can be use it.

Something like:

```
@OpenAPIRegister({Resource1.class, Resource2.class})
public class Api1 {
  public static void main(String[] args) {
    JoobyRunner.create(args)
      .bindResource(new Resource1())
      .bindResource(new Resource2())
      .run();
  }
}
```

> OpenAPIRegister
That or a better name.

So then the generator augment the resources found with what the annotation indicates.

**Comment by tpoll** [comment]
I do like the annotation route, the latest commit I pushed switches over to that style. I'm not sure what the best name is either. I added the word `manual` to try and make it clear this annotation is only needed when one is trying to manually register.

I'm not sure where the annotation should live -- should it still live in the openapi module or should it live in jooby core?

Where would you like to see the documentation? On the annotation itself, or in the doc links for openapi?

File: modules/jooby-openapi/src/main/java/io/jooby/openapi/OpenApiManualRegister.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+package io.jooby.openapi;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Target(ElementType.TYPE)
+@Retention(RetentionPolicy.RUNTIME)
+public @interface OpenApiManualRegister {
```
Comment (jknack) [inline modules/jooby-openapi/src/main/java/io/jooby/openapi/OpenApiManualRegister.java:15]: add documentation here.

File: modules/jooby-openapi/src/test/java/issues/i3059/App3059A.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ * Jooby https://jooby.io
+ * Apache License Version 2.0 https://jooby.io/LICENSE.txt
+ * Copyright 2014 Edgar Espina
+ */
+package issues.i3059;
+
+import io.jooby.Jooby;
+import io.jooby.openapi.OpenApiManualRegister;
+
+@OpenApiManualRegister({ControllerA3059.class, ControllerB3059.class})
```
Comment (jknack) [inline modules/jooby-openapi/src/test/java/issues/i3059/App3059A.java:11]: cool 😎

**Review by jknack — CHANGES_REQUESTED** [review]

**Comment by jknack** [comment]
> I'm not sure where the annotation should live -- should it still live in the openapi module or should it live in jooby core?

Let's add it on io.jooby.annotation package on jooby(-core)

**Comment by tpoll** [comment]
I believe I have addressed the feedback, I also thought a bit more and moved back to the originally proposed name of `OpenAPIRegister`

**Review by jknack — APPROVED** [review]

**Pull request merged**